tests/test_port.py

```diff
@@ -61,20 +62,51 @@ def test_kneighbors(result, n_components):
 )
 @pytest.mark.parametrize("n_components", [None, 3], ids=["full", "reduced"])
 @pytest.mark.parametrize("weighted", [False, True], ids=["unweighted", "weighted"])
-def test_predict(result, n_components, weighted):
+@pytest.mark.parametrize("training", [True, False], ids=["training", "testing"])
+def test_predict(result, n_components, weighted, training):
     """Test that the ported estimators predict the correct values."""
     method, estimator = result
     dataset = load_moscow_stjoes_results(method=method, n_components=n_components)
 
     weights = yaimpute_weights if weighted else None
-    trg_predicted = (
-        dataset.trg_predicted_weighted if weighted else dataset.trg_predicted_unweighted
+    validation_dataset = {
+        (True, True): dataset.ref_predicted_weighted,
+        (True, False): dataset.ref_predicted_unweighted,
+        (False, True): dataset.trg_predicted_weighted,
+        (False, False): dataset.trg_predicted_unweighted,
+    }
+    predicted = validation_dataset[(training, weighted)]
+
+    hyperparams = dict(n_neighbors=5, weights=weights)
+    hyperparams.update(
+        {"n_components": n_components} if hasattr(estimator(), "n_components") else {}
+    )
+    est = estimator(**hyperparams).fit(dataset.X_train, dataset.y_train)
+
+    prd = est.predict_independent() if training else est.predict(dataset.X_test)
+    assert_array_almost_equal(prd, predicted, decimal=3)
+
+
+@pytest.mark.uncollect_if(func=estimator_does_not_support_n_components)
+@pytest.mark.parametrize(
+    "result", ESTIMATOR_RESULTS.items(), ids=ESTIMATOR_RESULTS.keys()
+)
+@pytest.mark.parametrize("n_components", [None, 3], ids=["full", "reduced"])
+@pytest.mark.parametrize("weighted", [False, True], ids=["unweighted", "weighted"])
+def test_score_independent(result, n_components, weighted):
+    """Test that the ported estimators produce the correct score."""
+    method, estimator = result
+    dataset = load_moscow_stjoes_results(method=method, n_components=n_components)
+    weights = yaimpute_weights if weighted else None
+    predicted = (
+        dataset.ref_predicted_weighted if weighted else dataset.ref_predicted_unweighted
     )
+    expected_score = r2_score(dataset.y_train, predicted).mean()
```

Comment:
Rather than hard-code in scores to test against, I'm using the predicted files to create means of R^2 values for use in comparison.  I was slightly worried about speed, but it doesn't seem to be an issue.